src/main/java/org/cryptomator/ui/addvaultwizard/ChooseExistingVaultController.java

```diff
@@ -40,12 +42,12 @@ public class ChooseExistingVaultController implements FxController {
 	private final ObjectProperty<Vault> vault;
 	private final VaultListManager vaultListManager;
 	private final ResourceBundle resourceBundle;
-	private final Settings settings;
+	private final FxApplicationStyle applicationStyle;
 
-	private Image screenshot;
+	private final ObjectProperty<Image> screenshot = new SimpleObjectProperty<>();
```

Comment:
With the update to JFX 19, you can directly [create a mapping](https://openjfx.io/javadoc/19/javafx.base/javafx/beans/value/ObservableValue.html#map(java.util.function.Function)) in the constructor:
```java
private final ObservableValue<Image> screenshot;

ChooseExistingVaultController(....) {

    this.screenshot = applicationstyle.map(this::selectScreenshot)
}

private Image selectScreenshot(Theme theme) {
// create screenshot based on OS and theme
}

```

Reply:
Implemented in https://github.com/cryptomator/cryptomator/pull/2521/commits/2f9818aadec270b3ac0df9a5a1204e100e40226f